src/Http/Client.php

```diff
@@ -83,9 +91,12 @@ public function post(string $path, $body = null, array $query = [], ?string $con
     {
         if (!\is_null($contentType)) {
             $this->headers['Content-type'] = $contentType;
+        } elseif (str_ends_with($body, "}\n")) {
+            $this->headers['Content-type'] = 'application/x-ndjson';
+            $body = $this->json(new Ndjson)->serialize($body);
```

Comment:
I don't this is a good idea to switch format on the fly. Serializer should be preconfigured imho, or if contentType is manually set, then just instantiate other serializer without overriding the client state